### PR TITLE
#28 Use "Tag list" along with "sidecar files as tags for each image"

### PR DESCRIPTION
**Pull request by Seedmanc**
add manually entered tags to sidecar ones

### CODE REVIEW

File: js/uploader.js [hunk 1]
```diff
@@ -325,7 +325,7 @@ function RatingFor(file) {
 
 function TagsFor(file) {
     if (upOptions.tagging.when == 'sidecar')
-        return tagStorage[file.name] || ''
+        return NormTags(InfoAbout(file)[1].concat((tagStorage[file.name] || '').split(' ')));
```
Comment (Seedmanc) [inline js/uploader.js]: need to make sure there are no null pointer problems or that there won't be duplicating tags (if they can be a problem at all)

**Comment by ProximaNova** [comment]
I updated my fork to be up to date with your version. I then updates uploader.js with the above change. I redirected the appropriate URLs to URLs of my fork. (Check my updates if you think I messed up.) Unless GitHub takes a long time to realize my updates, it didn't work. "use sidecar files" along with "Tag list": "Tag list" is still greyed out.

2019-12-15 01:05:39 UTC - finished updating my fork.
2019-12-15 01:11:## UTC - uploaded 5 files (each with a sidecar .txt file):
[...]:11] Started uploading 5 files.
[...]:11] Image file1.jpeg was successfully uploaded.
[...]:11] Image file2.png was successfully uploaded.
[...]:11] Image file3.jpeg was successfully uploaded.
[...]:11] Image file4.png was successfully uploaded.
[...]:11] Image file5.png was successfully uploaded.
[...]:11] Finished uploading; 5 uploaded ok + 0 failed = 5 images total.

None of the sidecar files worked. Also, I made some bug reports yesterday; I think that it worked fine a month or multiple months ago, and by "it" I mean sidecar files were uploaded as tags for the image.

**Comment by Seedmanc** [comment]
Fixed, this should work.

**Comment by ProximaNova** [comment]
Just read https://github.com/Seedmanc/Booru-mass-uploader/pull/30#issuecomment-566740735 - I will test on Firefox and Google Chrome tomorrow.

I think you might have asked if sidecar files not working was an inconvenience in the past. I'd say that it is a big inconvenience the following took me like 30 minutes to 50 minutes. I had to see if it would work with a different browser, so I installed the malware known as Google Chrome. I couldn't install Google Chrome to the directory that I wanted. Therefore, I followed this guide: https://smallbusiness.chron.com/change-installation-path-google-chrome-42689.html <s>which is missing these commands (if they want to get users to save C:\ disk space, approx. 500 MB) which must have CMD ran as admin:
junction "C:\Program Files (x86)\Google\Chrome" "D:\user\Documents\ChromeSucks\ProgramFilesx86\Chrome"
junction "C:\Program Files (x86)\Google\CrashReports" "D:\user\Documents\ChromeSucks\ProgramFilesx86\CrashReports"
junction "C:\Program Files (x86)\Google\Policies" "D:\user\Documents\ChromeSucks\ProgramFilesx86\Policies"
junction "C:\Program Files (x86)\Google\Temp" "D:\user\Documents\ChromeSucks\ProgramFilesx86\Temp"
junction "C:\Program Files (x86)\Google\Update" "D:\user\Documents\ChromeSucks\ProgramFilesx86\Update"</s>

resulted in "Egads! Installation failed. Error code: 0x80070003." It made me uninstall then reinstall Google Chrome, and when I uninstalled it it asked that I fill out then submit a web page about why I uninstalled it. I told them I couldn't install it where I wanted to, that I didn't plan on reinstalling it, and that I would NOT recommend Chrome to friends and/or family. I guess Google didn't like my review because when I went to https://www.google.com/ it was the Swedish version of Google. I searched around with Google and looked at various parts of the google.com website for a while; I found that it could tell from my IP address that I am in the United States but it still gave me Swedish cuck shit:
sv (text I got): "[ZIP code], [city in a state in the USA] - Från din IP-adress - Använd exakt plats - Läs mer"
en (translation): "[ZIP code], [city in a state in the USA] - From your Internet address - Use precise location"

Why was I suddenly using swedish Google? I don't think I accidentally clicked anything. That was weird.

**Comment by Seedmanc** [comment]
Sorry if it's causing you trouble. This is why I made the /dev directory, where one can try and do some tests without affecting the working deployment, only need to point the scripts to that directory instead of the main one. 

Usually the branches are used for that, however, AFAIK there's only one branch available for github-pages, that enables the online functionality, so we have to make workarounds.

**Comment by ProximaNova** [comment]
https://github.com/Seedmanc/Booru-mass-uploader/pull/30/commits/7288ed03f8e1426d6538008c5c2818f943348fe7 fixed it, thanks! It works with Firefox 71.0 (64-bit); for example, it works with https://danbooru.donmai.us. I didn't test it with Google Chrome because it is not free open source software. Elitists/anime snobs might call this "prospecting", but you can find a lot of high quality anime images at https://ii.booru.org/index.php?page=post&s=list which are not at danbooru.donmai.us if you need to test the functionality with Danbooru websites. See if the image is already there with http://iqdb.org/.

**Pull request merged**